auto_nag/scripts/affected_flag_confirm.py

```diff
@@ -0,0 +1,91 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from libmozdata.bugzilla import Bugzilla
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+
+VERSION_PREFIX = "cf_status_firefox"
+N = len(VERSION_PREFIX)
+
+
+class AffectedFlagConfirm(BzCleaner):
+    def description(self):
+        return "Unconfirmed bugs that have an affected Firefox version flag"
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        data[bugid] = {"creator": bug["creator"]}
+        return bug
+
+    def get_bugs(self, *args, **kwargs):
+        bugs = super().get_bugs(*args, **kwargs)
+        self.query_url = None
+
+        def history_handler(bug, data):
+            bugid = str(bug["id"])
+            creator = data[bugid]["creator"]
+            removed_status = set()
+            for event in reversed(bug["history"]):
+                for change in event["changes"]:
+                    if (
+                        change["field_name"].startswith(VERSION_PREFIX)
+                        and change["field_name"] not in removed_status
+                    ):
+                        if change["added"] == "affected":
+                            if event["who"] != creator:
+                                affected_version = True
+                                break
+                        elif change["removed"] == "affected":
+                            removed_status.add(change["field_name"])
+                else:
+                    affected_version = False
+
+                if affected_version:
+                    break
+            else:
+                del data[bugid]
+
+        Bugzilla(
+            list(bugs.keys()),
+            historyhandler=history_handler,
+            historydata=bugs,
+            timeout=960,
+        ).get_data().wait()
+
+        return bugs
+
+    def get_bz_params(self, date):
+        fields = ["creator"]
+        params = {
+            "include_fields": fields,
+            "bug_status": "UNCONFIRMED",
+            "j1": "OR",
+            "f1": "OP",
+        }
+
+        last_version = utils.get_nightly_version_from_bz()
+        first_version = last_version - 40
+        for version in range(first_version, last_version + 1):
+            i = version - first_version + 2
+            params[f"f{i}"] = f"cf_status_firefox{version}"
+            params[f"o{i}"] = "equals"
+            params[f"v{i}"] = "affected"
```

Comment:
There are other statuses that we can consider equivalent to `affected`, e.g. `wontfix`, `fix-optional`, `disabled`, etc.

Reply:
I included `affected`, `wontfix`, `fix-optional`, `fixed`, `disabled`. However, I excluded `verified` and `verified-disabled` since they semes are not used.

Don in b1e13007864f791e2d15f5cde303e4ddf679d19b

Reply:
Let's add them too as they have been used in the past and might be used in the future